src/postgres/commands/createPostgresServer/steps/PostgresServerNameStep.ts

```diff
@@ -46,8 +47,7 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
         return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
     }
 
-    const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
-    const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
+    const availability: AbstractNameAvailability = (await client.checkNameAvailability(serverType, name));
```

Comment:
nit: not sure why those parentheses were there before
```suggestion
    const availability: AbstractNameAvailability = await client.checkNameAvailability(serverType, name);
```